src/cluster/kv/types.go

```diff
@@ -231,3 +221,35 @@ type TxnStore interface {
 
 	Commit([]Condition, []Op) (Response, error)
 }
+
+// PrefixStore is a multi-key store that supports prefix-based operations.
+// It exposes a Get() method that fetches all the values under the particular
+// prefix. It supports watching a prefix path and returns the current state
+// of the entire list of keys that are covered by the prefix.
+// It only supports read-only operations at the moment.
+type PrefixStore interface {
+	GetForPrefix(prefix string) (map[string]Value, error)
+	WatchForPrefix(prefix string) (PrefixWatch, error)
```

Comment:
non blocking: i think there's already a good amount of context that the operations here are for prefix based things, maybe calling these functions `GetValues` and `Watch` will suffice?

Reply:
:-) I wanted to do that but I'm reusing the same client struct that was used for satisfying the Store interface which already contains Get() and Watch().